File: admin/src/main/java/banner/dto/response/AdminBannerDetailResponseDto.java
package banner.dto.response;

import com.jigumulmi.banner.domain.Banner;
import java.time.LocalDateTime;
import lombok.Getter;
import lombok.experimental.SuperBuilder;

@Getter
@SuperBuilder
public class AdminBannerDetailResponseDto extends AdminBannerResponseDto {

    private LocalDateTime createdAt;
    private String outerImageS3Key;
    private String innerImageS3Key;

    public static AdminBannerDetailResponseDto from(Banner banner) {
        return AdminBannerDetailResponseDto.builder()
            .createdAt(banner.getCreatedAt())
            .modifiedAt(banner.getModifiedAt())
            .id(banner.getId())
            .title(banner.getTitle())
            .outerImageS3Key(banner.getOuterImageS3Key())
            .innerImageS3Key(banner.getInnerImageS3Key())
            .isActive(banner.getIsActive())
            .build();
    }
}

File: admin/src/main/kotlin/com/jigumulmi/banner/dto/response/AdminBannerDetailResponseDto.kt
package com.jigumulmi.banner.dto.response

import com.jigumulmi.banner.domain.Banner
import java.time.LocalDateTime

class AdminBannerDetailResponseDto(
    val createdAt: LocalDateTime,
    val modifiedAt: LocalDateTime,
    val id: Long,
    val title: String? = null,
    val isActive: Boolean? = false,
    val outerImageS3Key: String? = null,
    val innerImageS3Key: String? = null
    ) {

    companion object {
        fun from(banner: Banner): AdminBannerDetailResponseDto {
            return AdminBannerDetailResponseDto(
                createdAt = banner.createdAt,
                modifiedAt = banner.modifiedAt,
                id = banner.id,
                title = banner.title,
                isActive = banner.isActive,
                outerImageS3Key = banner.outerImageS3Key,
                innerImageS3Key = banner.innerImageS3Key,
            )
        }
    }
}
